### PR TITLE
Make `NETRC` env var handling aligned with GNU inetutils

### PR DESCRIPTION
**Pull request by Vyazovoy**
## Description

This PR changes the logic that handles `NETRC` env var value. According to the canonical [GNU inetutils implementation](https://github.com/guillemj/inetutils/blob/master/ftp/ruserpass.c#L120-L132) the env var value supposed to be used as is without appending any file names. So `/some_dir/abc.xyz` is a correct value and the `abc.xyz` file supposed to be parsed accordingly. To preserve backward compatibility with current behavior I keep appending the default netrc filename for cases when the env var points to a directory (that is incorrect according to `inetutils` implementation). So this implementation can be considered as a superset of the GNU `inetutils` plus some heuristics 😁

### CODE REVIEW

**Comment by Vyazovoy** [comment]
@beanieboi, @geemus, could you please take a look at this PR if you have some time

**Comment by geemus** [comment]
@Vyazovoy - unfortunately my permissions to the repository were removed at some point (probably at or around when I was laid off). I'm working on getting them reinstated, but for the time being I'm afraid I can't merge it and barring that I don't think review is going to be very useful.

**Comment by beanieboi** [comment]
> @Vyazovoy - unfortunately my permissions to the repository were removed at some point (probably at or around when I was laid off). I'm working on getting them reinstated, but for the time being I'm afraid I can't merge it and barring that I don't think review is going to be very useful.

@geemus I'd appreciate if you can do a review. I'm happy to push all the other buttons

**Review by geemus — APPROVED** [review]
LGTM, thanks!

**Comment by Vyazovoy** [comment]
@geemus, thanks for review and I'm sorry that you went through a layoff process 🫂 It is an awful event in any career.
@beanieboi, could you please approve the actions workflow and LMK if you need any changes.

**Comment by beanieboi** [comment]
@Vyazovoy can you please rebase? I just merged #60 which cleans up the ruby version we want to test

**Comment by Vyazovoy** [comment]
@beanieboi, sure, rebased ✅

**Review by beanieboi — APPROVED** [review]

**Comment by Vyazovoy** [comment]
@beanieboi, thanks for merging this PR 🙏 Would it be possible to make a `0.11.1` release (or maybe even `1.0` like @geemus suggested [here](https://github.com/heroku/netrc/pull/57#issuecomment-1902647089))? As I mentioned in #59 the last release was made in 2015 so it would be good to make a new one 😊

**Pull request merged**
